diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -23396,7 +23396,7 @@ else
 fi
 
 
-  pkg_config_args=gtk+-2.0
+  pkg_config_args="gtk+-2.0 hildon-1 hildon-fm-2"
   for module in .
   do
       case "$module" in
diff --git a/ui/gtk/fileselector.c b/ui/gtk/fileselector.c
--- a/ui/gtk/fileselector.c
+++ b/ui/gtk/fileselector.c
@@ -26,6 +26,7 @@
 #include <config.h>
 
 #include <gtk/gtk.h>
+#include <hildon/hildon.h>
 
 #include "gtkinternals.h"
 
@@ -45,17 +46,14 @@ run_dialog( const char *title, GtkFileChooserAction action )
   }
 
   dialog =
-    gtk_file_chooser_dialog_new( title, GTK_WINDOW( gtkui_window ),
-				 action, GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
-				 button, GTK_RESPONSE_ACCEPT,
-				 NULL );
+    hildon_file_chooser_dialog_new( GTK_WINDOW( gtkui_window ), action );
 
-  gtk_dialog_set_default_response( GTK_DIALOG( dialog ), GTK_RESPONSE_ACCEPT );
+  gtk_dialog_set_default_response( GTK_DIALOG( dialog ), GTK_RESPONSE_OK );
 
   if( current_folder )
     gtk_file_chooser_set_current_folder( GTK_FILE_CHOOSER( dialog ), current_folder );
 
-  if( gtk_dialog_run( GTK_DIALOG( dialog ) ) == GTK_RESPONSE_ACCEPT ) {
+  if( gtk_dialog_run( GTK_DIALOG( dialog ) ) == GTK_RESPONSE_OK ) {
     gchar *new_folder = gtk_file_chooser_get_current_folder( GTK_FILE_CHOOSER( dialog ) );
     if( new_folder ) {
       g_free( current_folder );
diff --git a/ui/gtk/gtkui.c b/ui/gtk/gtkui.c
--- a/ui/gtk/gtkui.c
+++ b/ui/gtk/gtkui.c
@@ -32,6 +32,7 @@
 #include <gtk/gtk.h>
 
 #include <glib.h>
+#include <hildon/hildon-program.h>
 
 #include <libspectrum.h>
 
@@ -116,7 +117,7 @@ ui_init( int *argc, char ***argv )
   gtk_widget_set_default_colormap( gdk_rgb_get_cmap() );
   gtk_widget_set_default_visual( gdk_rgb_get_visual() );
 
-  gtkui_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+  gtkui_window = hildon_window_new();
 
   gtk_window_set_title( GTK_WINDOW(gtkui_window), "Fuse" );
   gtk_window_set_wmclass( GTK_WINDOW(gtkui_window), fuse_progname, "Fuse" );
@@ -145,7 +146,7 @@ ui_init( int *argc, char ***argv )
 		   gtkui_menu_data_size );
 
   gtk_window_add_accel_group( GTK_WINDOW(gtkui_window), accel_group );
-  gtk_box_pack_start( GTK_BOX(box), menu_bar, FALSE, FALSE, 0 );
+  hildon_window_set_menu (HILDON_WINDOW (gtkui_window), GTK_MENU (menu_bar));
   
   gtkui_drawing_area = gtk_drawing_area_new();
   if(!gtkui_drawing_area) {
@@ -220,7 +221,7 @@ static gboolean gtkui_make_menu(GtkAccelGroup **accel_group,
 				guint menu_data_size)
 {
   *accel_group = gtk_accel_group_new();
-  menu_factory = gtk_item_factory_new( GTK_TYPE_MENU_BAR, "<main>",
+  menu_factory = gtk_item_factory_new( GTK_TYPE_MENU, "<main>",
 				       *accel_group );
   gtk_item_factory_create_items( menu_factory, menu_data_size, menu_data,
 				 NULL);
