diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -23396,7 +23396,7 @@ else
 fi
 
 
-  pkg_config_args="gtk+-2.0 hildon-1 hildon-fm-2"
+  pkg_config_args="gtk+-2.0 hildon-1 hildon-fm-2 libosso"
   for module in .
   do
       case "$module" in
diff --git a/ui/gtk/gtkui.c b/ui/gtk/gtkui.c
--- a/ui/gtk/gtkui.c
+++ b/ui/gtk/gtkui.c
@@ -33,6 +33,7 @@
 
 #include <glib.h>
 #include <hildon/hildon-program.h>
+#include <libosso.h>
 
 #include <libspectrum.h>
 
@@ -111,6 +112,7 @@ ui_init( int *argc, char ***argv )
   GdkWindowHints hints;
 
   gtk_init(argc,argv);
+  osso_initialize ("org.maemo.fuse", VERSION, FALSE, NULL);
 
   gdk_rgb_init();
   gdk_rgb_set_install( TRUE );
