diff --git a/ui/gtk/gtkui.c b/ui/gtk/gtkui.c
--- a/ui/gtk/gtkui.c
+++ b/ui/gtk/gtkui.c
@@ -112,6 +112,7 @@ ui_init( int *argc, char ***argv )
   GdkWindowHints hints;
 
   gtk_init(argc,argv);
+  g_set_application_name ("Fuse");
   osso_initialize ("org.maemo.fuse", VERSION, FALSE, NULL);
 
   gdk_rgb_init();
@@ -122,7 +123,6 @@ ui_init( int *argc, char ***argv )
   gtkui_window = hildon_window_new();
 
   gtk_window_set_title( GTK_WINDOW(gtkui_window), "Fuse" );
-  gtk_window_set_wmclass( GTK_WINDOW(gtkui_window), fuse_progname, "Fuse" );
 
   gtk_window_set_default_size( GTK_WINDOW(gtkui_window),
 			       DISPLAY_ASPECT_WIDTH, DISPLAY_SCREEN_HEIGHT );
