diff --git a/ui/gtk/gtkkeyboard.c b/ui/gtk/gtkkeyboard.c
--- a/ui/gtk/gtkkeyboard.c
+++ b/ui/gtk/gtkkeyboard.c
@@ -82,6 +82,9 @@ gtkkeyboard_keypress( GtkWidget *widget GCC_UNUSED, GdkEvent *event,
   fuse_event.type = INPUT_EVENT_KEYPRESS;
   get_keysyms( &fuse_event, event->key.hardware_keycode, event->key.keyval );
 
+  if (event->key.state == GDK_MOD5_MASK)
+    fuse_event.types.key.spectrum_key = fuse_event.types.key.native_key;
+
   return input_event( &fuse_event );
 
   /* FIXME: handle F1 to deal with the pop-up menu */
@@ -96,5 +99,8 @@ gtkkeyboard_keyrelease( GtkWidget *widget GCC_UNUSED, GdkEvent *event,
   fuse_event.type = INPUT_EVENT_KEYRELEASE;
   get_keysyms( &fuse_event, event->key.hardware_keycode, event->key.keyval );
 
+  if (event->key.state == GDK_MOD5_MASK)
+    fuse_event.types.key.spectrum_key = fuse_event.types.key.native_key;
+
   return input_event( &fuse_event );
 }
diff --git a/ui/gtk/keysyms.c b/ui/gtk/keysyms.c
--- a/ui/gtk/keysyms.c
+++ b/ui/gtk/keysyms.c
@@ -37,6 +37,7 @@ keysyms_map_t keysyms_map[] = {
 
   { GDK_Tab,         INPUT_KEY_Tab         },
   { GDK_Return,      INPUT_KEY_Return      },
+  { GDK_KP_Enter,    INPUT_KEY_Return      },
   { GDK_space,       INPUT_KEY_space       },
   { GDK_numbersign,  INPUT_KEY_numbersign  },
   { GDK_apostrophe,  INPUT_KEY_apostrophe  },
