From: Alberto Garcia <agarcia@igalia.com>
Subject: Don't use the SDL audio driver, it has issues with pulseaudio
Bug: http://sourceforge.net/tracker/?func=detail&aid=3483869&group_id=91293&atid=596648
Forwarded: no
Index: fuse-emulator/configure.in
===================================================================
--- fuse-emulator.orig/configure.in
+++ fuse-emulator/configure.in
@@ -458,10 +458,7 @@ dnl Decide which sound routines to use
 dnl
 
 AC_MSG_CHECKING(which sound routines to use)
-if test "$UI" = sdl; then
-  SOUND_LIBADD='sdlsound.$(OBJEXT)' SOUND_LIBS='' sound_fifo=yes
-  AC_MSG_RESULT(SDL)
-elif test "$dxsound_available" = yes; then
+if test "$dxsound_available" = yes; then
   SOUND_LIBADD='dxsound.$(OBJEXT)' SOUND_LIBS='-ldsound -lole32 -ldxguid'
   AC_MSG_RESULT(DirectX)
   AC_DEFINE([DIRECTSOUND_VERSION], 0x0700, [DirectX 7 or higher is required])
