From: Alberto Garcia <berto@igalia.com>
Subject: Add bash completion for the fuse-gtk and fuse-sdl binaries
Forwarded: not-needed
Index: fuse-emulator/data/shell-completion/bash/fuse
===================================================================
--- fuse-emulator.orig/data/shell-completion/bash/fuse
+++ fuse-emulator/data/shell-completion/bash/fuse
@@ -284,4 +284,4 @@ _fuse()
         _filedir '@(csw|CSW|ltp|LTP|pzx|PZX|spc|SPC|sta|STA|tap|TAP|tzx|TZX|wav|WAV|slt|SLT|sna?(pshot)|SNA?(PSHOT)|?(mgt)snp|?(MGT)SNP|sp|SP|szx|SZX|z80|Z80|zxs|ZXS|d40|D40|d80|D80|dsk|DSK|img|IMG|fdi|FDI|mgt|MGT|opd|OPD|opu|OPU|sad|SAD|scl|SCL|td0|TD0|trd|TRD|udi|UDI|rzx|RZX|dck|DCK|hdf|HDF|mdr|MDR)'
     fi
 } &&
-complete -F _fuse -o filenames fuse
+complete -F _fuse -o filenames fuse fuse-gtk fuse-sdl
