From: Alberto Garcia <berto@igalia.com>
Subject: Prefer the X11 GDK backend over the Wayland one
Bug: https://sourceforge.net/p/fuse-emulator/bugs/367/
Bug-Debian: https://bugs.debian.org/872994
Index: fuse-emulator/ui/gtk/gtkui.c
===================================================================
--- fuse-emulator.orig/ui/gtk/gtkui.c
+++ fuse-emulator/ui/gtk/gtkui.c
@@ -153,6 +153,11 @@ ui_init( int *argc, char ***argv )
   GtkAccelGroup *accel_group;
   GtkSettings *settings;
 
+#if GTK_CHECK_VERSION( 3, 10, 0 )
+  /* The Wayland output is buggy, see #367 */
+  gdk_set_allowed_backends( "quartz,win32,mir,x11,*" );
+#endif
+
   gtk_init(argc,argv);
 
 #if !GTK_CHECK_VERSION( 3, 0, 0 )
