From: Alberto Garcia <berto@igalia.com>
Subject: Show the Fuse icon on the GTK window and About dialog
Bug: https://sourceforge.net/p/fuse-emulator/patches/413/
Index: fuse-emulator/ui/gtk/gtkui.c
===================================================================
--- fuse-emulator.orig/ui/gtk/gtkui.c
+++ fuse-emulator/ui/gtk/gtkui.c
@@ -169,6 +169,8 @@ ui_init( int *argc, char ***argv )
 
   gtkui_window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
 
+  gtk_window_set_icon_name( GTK_WINDOW( gtkui_window ), "fuse" );
+
   settings = gtk_widget_get_settings( GTK_WIDGET( gtkui_window ) );
   g_object_set( settings, "gtk-menu-bar-accel", "F1", NULL );
   gtk_window_set_title( GTK_WINDOW(gtkui_window), "Fuse" );
@@ -672,7 +674,7 @@ menu_help_about( GtkAction *gtk_action G
                          "program-name", "Fuse",
                          "comments", "The Free Unix Spectrum Emulator",
                          "copyright", FUSE_COPYRIGHT,
-                         "logo-icon-name", NULL,
+                         "logo-icon-name", "fuse",
                          "version", VERSION,
                          "website", PACKAGE_URL,
                          NULL );
